.grit/patterns/js/no_implicit_coercion.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Disallow shorthand type conversions
+tags: [good-practice]
+---
+This matches the [eslint rule](https://eslint.org/docs/latest/rules/no-implicit-coercion).
+
+Check the [Playground](https://app.grit.io/studio?key=cle6qCCw_-XMW9Nxs-PKH)
+```grit
+engine marzano(1.0)
+language js
+
+function parenthesized_or_not($exp){
+     or {
+         and {
+             $exp <: parenthesized_expression(),
+             return $exp
+         },
+         and {
+             return `($exp)`
+         }
+     },
+ }
+
+ or {
+     `!!$v` as $exp where {
+         $e = parenthesized_or_not($v),
+         $exp => `Boolean$e`
+     },
+     `~$v` as $exp where {
+         $e = parenthesized_or_not($v),
+         $exp => `$e !== -1`
+     },
+     or {
+         `+$v`,
+         `-(-$v)`,
+         `$v - 0`,
+         `1 * $v`,
+         `$v * 1`
+     } as $exp where {
+         $e = parenthesized_or_not($v),
+         $exp => `Number$e`
+     },
+     or {
+     `"" + $v`,
+     `$v + ""`,
+     } as $exp where {
+         $e = parenthesized_or_not($v),
+         $exp => `String$e`
+     },
+ }
+
+```
+
+Code examples:
```

Comment:
Please look at the other patterns in this repo to learn the format for examples/testing. You need a heading above the examples.

```suggestion
## Code examples
```